File: CS/MvcApp_TreeView/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;

namespace MvcApp_TreeView.Controllers {
    public class HomeController : Controller {
        public object CheckedValues { get; set; }

        public ActionResult Index() {
            ViewBag.Message = "Welcome to DevExpress Extensions for ASP.NET MVC!";
            return View(CheckedValues);
        }

        [HttpPost]
        public ActionResult PostTreeViewValues() {
            CheckedValues = Request.Params["hf"];

            ViewBag.Message = "The TreeView state has been restored successfully";
            return View("Index", CheckedValues);
        }
    }
}

File: VB/MvcApp_TreeView/Controllers/HomeController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc

Namespace MvcApp_TreeView.Controllers
	Public Class HomeController
		Inherits Controller
		Private privateCheckedValues As Object
		Public Property CheckedValues() As Object
			Get
				Return privateCheckedValues
			End Get
			Set(ByVal value As Object)
				privateCheckedValues = value
			End Set
		End Property

		Public Function Index() As ActionResult
			ViewBag.Message = "Welcome to DevExpress Extensions for ASP.NET MVC!"
			Return View(CheckedValues)
		End Function

		<HttpPost> _
		Public Function PostTreeViewValues() As ActionResult
			CheckedValues = Request.Params("hf")

			ViewBag.Message = "The TreeView state has been restored successfully"
			Return View("Index", CheckedValues)
		End Function
	End Class
End Namespace
